File: Coderbyte/javascript/simple-adding/simple_adding.js
// For this challenge you will be adding up all the numbers from 1 to a certain argument.

module.exports = {
  // Power sum
  simpleAdding: function(n){
    return (n*(n+1))/2;
  },
  // NOTE: This is a substantially slower method
  // It is only here for demonstration purposes
  simpleAddingBruteForce: function(n){
    var total = 0;
    for (var i = 1; i < n + 1; i++)
      total = total + i
    return total;
  }
};

File: Coderbyte/ruby/simple-adding/simple_adding.rb
#!/usr/bin/env ruby

require 'benchmark'

# For this challenge you will be adding up all the numbers from 1 to a certain argument.

# Power sum
def simple_adding(n)
  (n*(n+1))/2
end

# Alternative brute force method
# def simple_adding(n)
#   total = 0
#   1.upto(n) do |i|
#     total = total + i
#   end
#   total
# end

puts simple_adding(1)
puts simple_adding(2)
puts simple_adding(5)
puts simple_adding(10)
puts simple_adding(20)
puts simple_adding(400)